packages/contracts-bedrock/test/DelayedVetoable.t.sol

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { CommonTest, Reverter } from "./CommonTest.t.sol";
+import { DelayedVetoable } from "../src/L1/DelayedVetoable.sol";
+
+contract DelayedVetoable_Init is CommonTest {
+    error Unauthorized(address expected, address actual);
+    error ForwardingEarly();
+
+    event Initiated(bytes32 indexed callHash, bytes data);
+    event Forwarded(bytes32 indexed callHash, bytes data);
+    event Vetoed(bytes32 indexed callHash, bytes data);
+
+    address target = address(0xabba);
+    address initiator = alice;
+    address vetoer = bob;
+    uint256 operatingDelay = 14 days;
+    DelayedVetoable delayedVetoable;
+    Reverter reverter;
+
+    function setUp() public override {
+        super.setUp();
+        delayedVetoable = new DelayedVetoable({
+            initiator_: alice,
+            vetoer_: bob,
+            target_: address(target),
+            operatingDelay_: operatingDelay
+        });
+        // Most tests will use the operating delay, so we call as the initiator with null data
+        // to set the delay. For tests that need to use the initial zero delay, we'll modify the
+        // value in storage.
+        vm.prank(initiator);
+        (bool success,) = address(delayedVetoable).call(hex"");
+
+        reverter = new Reverter();
+    }
+
+    /// @dev This function is used to prevent initiating the delay unintentionally..
+    /// @param data The data to be used in the call.
+    function assumeNonzeroData(bytes memory data) internal pure {
+        vm.assume(data.length > 0);
+    }
+}
+
+contract DelayedVetoable_Getters_Test is DelayedVetoable_Init {
+    /// @dev The getters return the expected values when called by the zero address.
+    function test_getters() external {
+        vm.startPrank(address(0));
+        assertEq(delayedVetoable.initiator(), initiator);
+        assertEq(delayedVetoable.vetoer(), vetoer);
+        assertEq(delayedVetoable.target(), target);
+        assertEq(delayedVetoable.delay(), operatingDelay);
+    }
+}
+
+contract DelayedVetoable_Getters_TestFail is DelayedVetoable_Init {
+    function test_getters_notVetoer() external {
+        // getter calls from addresses other than the zero address will revert in the
+        // initiation branch of the proxy.
+        vm.assume(msg.sender != address(0) && msg.sender != initiator && msg.sender != vetoer);
```

Comment:
What does `vm.assume` do in a non fuzz test?

Reply:
oh nothing, that's a bug good call out.